docs/manual/spec.md

```diff
@@ -461,6 +461,13 @@ when name is omitted, the description refers to the main package.
 Package build is divided into multiple separate steps, each executed
 in a separate shell.
 
+Only one of each section can be present in a spec, but all build scriptlets
+except for `%prep` accept options `-a` and `-p`, for append and prepend mode.
+Append and prepend append or prepend lines to the section in the order they
+appear in the spec. Both append and prepend can be used multiple times and
+without other restrictions, but a section without either mode can only
+appear first (eg `%build` cannot follow `%build -p`).
```

Comment:
This feels artificial to me.

Not mentioning that the attached test case does not contain `%install` section as far as I can see. So how it is supposed to work in this case?

Reply:
As it says, append and prepend can be used without any restrictions. So if eg %install does not already exist then '%install -a' or '%install -p' is simply equivalent to '%install'. However plain '%install' can only appear first because anything else would be ambiguous.

Reply:
I mentioned the reason in the new version of the manual.

Reply:
Turns out what felt artificial, was in fact an artificial limit of the implementation. 
https://github.com/rpm-software-management/rpm/pull/3047 does away with that limitation, -a/-p are always applied relative to the main section if it exists, otherwise the first fragment is used as the base. Makes a lot more sense that way.